### ISSUE TITLE
Supabase Realtime not working with table name

### ISSUE THREAD

**Issue by RentfireFounder** [supabase/realtime-js#254]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

Supabase Realtime doesn't get emit events when provided with table name, but works without table name in localhost

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Went to `/database/replication` and turned on `supabase_realtime`

<img width="1500" alt="Screenshot 2023-10-10 at 3 07 12 PM" src="https://github.com/supabase/supabase/assets/127445161/906754aa-f058-41f6-ad3e-f4c08dbde801">

2. Turned on table for which I want realtime events 

<img width="1228" alt="Screenshot 2023-10-10 at 3 07 20 PM" src="https://github.com/supabase/supabase/assets/127445161/23bf740a-3c3d-4376-aa08-5574872a50d9">

In Javascript, If I do something like this, this won't work

```
supabase
            .client 
            .channel('user_list_changes')
            .on(
                'postgres_changes',
                {
                    event: '*',
                    schema: 'public',
                    table: 'users_list',
                },
                (payload: any) => console.log('payload', payload),
            )
            .subscribe();
 ```
 
 but this would 
 
 ```
 supabase
            .client 
            .channel('user_list_changes')
            .on(
                'postgres_changes',
                {
                    event: '*',
                    schema: 'public',
                },
                (payload: any) => console.log('payload', payload),
            )
            .subscribe();
   ```
## Expected behavior

I was expecting to receive events when provided with `table_name`

## System information

- OS: Mac
- Browser: Brave
- Version of supabase-js: 
```
   "@supabase/supabase-js@^2.10.0":
   "@supabase/functions-js" "^2.1.0"
    "@supabase/gotrue-js" "^2.46.1"
    "@supabase/postgrest-js" "^1.8.0"
    "@supabase/realtime-js" "^2.7.4"
    "@supabase/storage-js" "^2.5.1"
    ```
- Version of Node.js: 18.3.0

**Comment by RentfireFounder** [comment]
@kamilogorek 

Both of these don't log on update when I have enabled RLS policy on table

```
    React.useEffect(() => {
        console.log('here');
        serviceSubscription.current = supabase
            .getClient()
            .channel(`service-data-${serviceName}`)
            .on(
                'postgres_changes',
                {
                    event: '*',
                    schema: 'public',
                    table: 'services_connected',
                },
                (payload) => console.log('Change', payload),
            )
            .subscribe();

        supabase
            .getClient()
            .channel(`service-data`)
            .on(
                'postgres_changes',
                {
                    event: '*',
                    schema: 'public',
                },
                (payload) => console.log('Global', payload),
            )
            .subscribe();
    }, []);
  ```
    
  Here is my RLS policy. (Normal Select queries work as expected). 
  
<img width="1512" alt="Screenshot 2023-10-21 at 5 52 35 AM" src="https://github.com/supabase/realtime-js/assets/127445161/cef2f00f-548d-4601-8aad-61f1a2ba7675">

    
<img width="1512" alt="Screenshot 2023-10-21 at 5 52 21 AM" src="https://github.com/supabase/realtime-js/assets/127445161/7844a3bf-115d-4840-87c1-68d21e02e1b9">

**Comment by RentfireFounder** [comment]
@kamilogorek Any update on this?

**Comment by filipecabaco** [comment]
@RentfireFounder it's working locally but not working when using Supabase infrastructure?

**Comment by RentfireFounder** [comment]
@filipecabaco no it's working when RLS is disabled but not working when RLS is enabled.

With RLS enabled, the select/update and normal queries work but realtime doesn't

if this is the problem with my RLS rules, why is normal queries working and not realtime? 

Also, more on this comment: https://github.com/supabase/realtime-js/issues/254#issuecomment-1773552145

**Comment by w3b6x9** [comment]
@RentfireFounder just to confirm you're spinning up the Supabase stack locally correct? This is a locally running version of Realtime that RLS is not working for you. Is that right?

**Comment by RentfireFounder** [comment]
@w3b6x9 Yap!

**Comment by RentfireFounder** [comment]
@w3b6x9 @filipecabaco @kamilogorek were you able to replicate it? any updatea?

**Comment by victorbillaud** [comment]
Hello, i got the same error...

**Comment by roxxid** [comment]
Exact same thing is happening with me, (working with only prod) with RLS turned on i only get DELETE events. No matter what RLS policy I add I dont see any events coming in. Like the OP said on turning RLS off again postgres changes start flowing again.

**Comment by RentfireFounder** [comment]
@filipecabaco @w3b6x9

**Comment by filipecabaco** [comment]
hey everyone, sorry we were working on a lot of changes and were not able to dedicate enough time to close this issue.

is this issue persisted? were you able to tackle it?

**Comment by RentfireFounder** [comment]
@filipecabaco nope, Also, just in case, did you see this comment? could it be because I am creating my own jwt and using that>

**Comment by filipecabaco** [comment]
We do have an issue currently with custom jwts where the errors are not properly shown to the user 😞 I wonder if this is related 🤔

**Comment by rogaha** [comment]
> Exact same thing is happening with me, (working with only prod) with RLS turned on i only get DELETE events. No matter what RLS policy I add I dont see any events coming in. Like the OP said on turning RLS off again postgres changes start flowing again.
> 
> UPDATE: I added a SELECT policy with anon role set as TRUE (not recommended but it worked for now and can carry on with the dev)

I'm having the same issue. Any updates here?

**Comment by sidtuladhar** [comment]
having the same issue :(

**Comment by therealtimhawkins** [comment]
Same for me, anyone had any luck?

**Comment by andrew-wang-polar** [comment]
I'm also experiencing this issue. I'm using an RLS policy that checks a custom JWT created using the supabase jwt secret (with the user matching a third party auth token). My RLS policy is working fine with regular requests but not realtime requests.

When I turn off the RLS (or just set it to USING (true)) I get all the events. When I have it on I only get DELETE events.

**Comment by andrew-wang-polar** [comment]
I solved the issue. See this article: https://liviogamassia.medium.com/using-supabase-rls-with-firebase-auth-custom-auth-provider-357eaad9c70f

In my case I was already correctly setting role and aud = 'authenticated', but the thing I needed to do was also use `realtime.setAuth(supabaseJwt)`:

```
const supabaseJwt = generateSupabaseJwt(
  supabaseJwtInfo,
  SUPABASE_JWT_SECRET
);
const supabaseClient = createClient<Database>(
  SUPABASE_URL,
  SUPABASE_ANON_KEY,
  {
    auth: {
      persistSession: false,
      autoRefreshToken: false,
      detectSessionInUrl: false,
    },
    global: {
      headers: {
        Authorization: `Bearer ${supabaseJwt}`,
      },
    },
  }
);
supabaseClient.realtime.setAuth(supabaseJwt);
```

**Comment by maxtuzz** [comment]
Wondering if my issues are something to do with this. Since updating 

`"@supabase/supabase-js": "2.53.0"` -> `"@supabase/supabase-js": "2.57.4",` which included a few real time changes from this package, my realtime subscriptions broke 9/10 times. Occasionally they would work surprisingly but most of the time they wouldn't. 

Downgrading my package again fixed the issue. 

I have realtime all throughout my app and have a base hook that higher-order hooks leverage:

```
const realtimeState = useRealtimeBase({
    config: {
      channelName: `item:${id}:items`,
      table: 'my_table',
      event: RealtimeEvent.All,
      filter: `myId=eq.${myId}`,
      enabled: enabled && !!myId,
      debugLabel: '...',
    },
    onPayload: handlePayload,
    dependencies: [threadId],
  });
```

and in my realtime base I have something like 

```
const channel = supabase
        .channel(channelName)
        .on(
          'postgres_changes',
          {
            schema: 'public',
            table,
            event,
            ...(filter && { filter }),
          },
          (payload: RealtimePostgresChangesPayload<Record<string, object>>) => {
            enableLogging && log('Received payload:', payload);
            onPayloadRef.current?.(payload);
          },
        )
        .subscribe(handleStatus);

      channelRef.current = channel;
    };
```
